### PR TITLE
[IBCDPE-798] Implements GX Expectation Suite for `team_info` Dataset

### PR DESCRIPTION
**Pull request by BWMac**
Description:

This PR implements a GX expectation suite for the `team_info` dataset.

Please let me know if there are any additional expectations I should add, or if there are any adjustments to be made to the expectations I have already implemented.

Notes:
- This change has been tested by running `adt test_config.yaml`. Example results can be downloaded [here](https://www.synapse.org/#!Synapse:syn53618618).
- I generated `members_schema.json` using [this](https://jsonschema.net/) Schema Generator and then added a few more details.

### CODE REVIEW

File: gx_suite_definitions/team_info.ipynb [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# description\n",
+    "validator.expect_column_values_to_be_of_type(\"description\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"description\")\n",
+    "validator.expect_column_values_to_match_regex(\"description\", \"^[a-zA-Z0-9-(),.':/ ]+$\")\n",
```
Comment (jaclynbeck-sage) [inline gx_suite_definitions/team_info.ipynb:147]: Nit: Are there any characters that are NOT allowed in this field? i.e. do we need a regex at all?

File: gx_suite_definitions/team_info.ipynb [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# members\n",
+    "validator.expect_column_values_to_be_of_type(\"members\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"members\")\n",
+    "#get JSON schema\n",
+    "with open(\"../src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json\", \"r\") as file:\n",
+    "    gene_biodomains_schema = json.load(file)\n",
+    "validator.expect_column_values_to_match_json_schema(\"members\", json_schema=gene_biodomains_schema)"
```
Comment (jaclynbeck-sage) [inline gx_suite_definitions/team_info.ipynb]: Rename from `gene_biodomains_schema` -> `members_schema`

File: src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "items": {
+        "type": "object",
+        "required": [
+            "isprimaryinvestigator",
+            "name",
+            "url"
+        ],
+        "properties": {
+            "isprimaryinvestigator": {
+                "type": "boolean"
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json:16]: Any way to check that this field isn't null?

File: src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            },
+            "name": {
+                "type": "string"
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json:19]: Any way to check that this field isn't null?

File: gx_suite_definitions/team_info.ipynb (hunk 1 above)
Comment (BWMac) [reply gx_suite_definitions/team_info.ipynb:147]: I think this guards against non UTF-8 characters, but if it isn't necessary I am happy to remove it

File: src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json (hunk 3 above)
Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json:16]: From my understanding, the type field enforces that the value cannot be `null`. To allow `null` values, you would have to do something like [this](https://stackoverflow.com/questions/16241333/specify-a-value-can-be-a-string-or-null-with-json-schema).

File: gx_suite_definitions/team_info.ipynb (hunk 2 above)
Comment (BWMac) [reply gx_suite_definitions/team_info.ipynb]: changed in latest commit

File: src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json (hunk 3 above)
Comment (jaclynbeck-sage) [reply src/agoradatatools/great_expectations/gx/json_schemas/team_info/members_schema.json:16]: Ok cool, thanks.

File: gx_suite_definitions/team_info.ipynb (hunk 1 above)
Comment (jaclynbeck-sage) [reply gx_suite_definitions/team_info.ipynb:147]: @JessterB thoughts?

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!

File: gx_suite_definitions/team_info.ipynb (hunk 1 above)
Comment (JessterB) [reply gx_suite_definitions/team_info.ipynb:147]: We have historically had problems with UTF-8 characters (mainly Microsoft "smart" quotes, and alpha/beta/etc. characters) in user-submitted content like team descriptions, nominations, etc. These get transformed into "� (REPLACEMENT CHARACTER, [U+FFFD]" during processing. Ideally we will eventually detect the unsupported characters during source file validation (unless anyone can think of a way to improve ADT to handle them?), and ADT should never see them. We aren't there yet though, so catching these before we display them in Agora would be a good thing. One option would be to change the regex to just specifically disallow U+FFFD. Or we could wait to see if the current regex is problematic before trying to adjust it.

File: src/agoradatatools/great_expectations/gx/expectations/team_info.json [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+      },
+      "meta": {}
+    },
+    {
+      "expectation_type": "expect_column_values_to_be_in_set",
+      "kwargs": {
+        "column": "program",
+        "value_set": [
+          "AMP-AD",
+          null,
```
Comment (JessterB) [inline src/agoradatatools/great_expectations/gx/expectations/team_info.json:114]: Can we add a multi-field rule to ensure that  if program == null, team == "Sage Bionetworks"? No other team should be allowed to have a null program value...

**Review by JessterB — CHANGES_REQUESTED** [review]

File: src/agoradatatools/great_expectations/gx/expectations/team_info.json (hunk 5 above)
Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/expectations/team_info.json:114]: I will take a look at this today

File: gx_suite_definitions/team_info.ipynb (hunk 1 above)
Comment (BWMac) [reply gx_suite_definitions/team_info.ipynb:147]: I think what I have here _should_ prevent the issues you are describing. So maybe we just see how it goes?

File: src/agoradatatools/great_expectations/gx/expectations/team_info.json (hunk 5 above)
Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/expectations/team_info.json:114]: @JessterB At first glance, it doesn't look like there is an existing expectation that would do what you are describing. Is this worth creating a custom expectation (~1 sp of work)?

Comment (JessterB) [reply src/agoradatatools/great_expectations/gx/expectations/team_info.json:114]: @BWMac Eh, not really. We don't add new teams all that often, and when we do there is already enough manual handling and validation required that we really should catch a missing program.  Let's not spend a point on it for now.

File: gx_suite_definitions/team_info.ipynb (hunk 1 above)
Comment (JessterB) [reply gx_suite_definitions/team_info.ipynb:147]: Sounds reasonable to me.

**Review by JessterB — APPROVED** [review]

**Review by jaclynbeck-sage — APPROVED** [review]

File: src/agoradatatools/great_expectations/gx/expectations/team_info.json (hunk 5 above)
Comment (JessterB) [reply src/agoradatatools/great_expectations/gx/expectations/team_info.json:114]: tracking this via AG-1391 for the future

**Pull request merged**